### PR TITLE
refactor(app): add choose protocol slideout to devices t1 and t2

### LINKED ISSUE #8675: Devices Landing: "Why is my robot not available?" modal
# Design
Unavailable robots will have an option in their three-dot menu to learn why their robot is unavailable. See how this looks [here](https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/%5Bworking%5D-Milestone-1%3A-Read-and-Execute?node-id=1297%3A88551). Clicking on that menu item will open the "Why is my robot not available?" modal

# Acceptance Criteria

- [ ] Unavailable robots have an option in their three-dot menu to open this modal 

- [ ] The modal renders correctly

- [ ] "Learn more about troubleshooting connection problems" links to this page: 
https://support.opentrons.com/en/articles/2687601-troubleshooting-connection-problems

- [ ] Users can close the modal by clicking the 'x', the close button, or anywhere outside the modal

### LINKED ISSUE #8803: Run a protocol - Choose a Protocol Slide-out
# Overview
Clicking the "Run a Protocol" button on the Device Details page will open up this "Choose a Protocol" slide-out. Here a user can select the protocol they'd like to run on the robot and then advance to the setup tab on the protocol run details page. 

**Update from 3/10**: We're simplifying this slide-out for 5.1 by removing the protocol sorting and the robot configuration information. Both features may return in an upcoming release. 

# Design
https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=1553%3A106633

# Acceptance Criteria

- [ ] The user is able to select a protocol and proceed to a new protocol run details page

- [ ] Protocols on a software version ahead of the robot software version should not be selectable

- [ ] Clicking on the "Proceed to Setup" button should advance the user to the setup tab on the protocol run details page

### LINKED ISSUE #9784: Robot card: Add a link to the robot settings
# Overview
Since the robot settings are buried a few levels deep, we should add a direct link to the robot settings page from each robot card on the devices landing and device details pages. 

# Design
The Robot Settings link should be the bottom one in the overflow menu as seen here: https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=1297%3A91474

# Acceptance Criteria

- [ ] Each robot card on the devices landing contains an overflow menu link to the robot settings page

- [ ] Each robot card on the device details page contains an overflow menu link to the robot settings page

### LINKED ISSUE #9897: Simplify device card behavior
# Overview
Let's simplify the robot card behavior so that the labware cards, protocol cards, and device cards all work the same way

# Implementation Details
- Clicking on the device card should lead to the device details page
- Remove the option to see robot details from the device card's overflow menu

# Acceptance Criteria
 
- [ ] The device details page can be accessed by clicking on the device card

- [ ]  The device details cannot be accessed via the overflow menu

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Provide slideout view to allow users to choose a protocol to run on a given robot, either from the
overflow menu of a specific robot card on the devices landing page, or from the run protocol button
on a robots detail page.

Closes https://github.com/Opentrons/opentrons/issues/9784, Closes https://github.com/Opentrons/opentrons/issues/9897, Closes https://github.com/Opentrons/opentrons/issues/8675, Closes https://github.com/Opentrons/opentrons/issues/8803

TODO:

- [x] empty state for the `ChooseProtocolSlideout` component
- [x] unit tests for the `ChooseProtocolSlideout` component
- [x] test coverage for `Slideout` overlay click to close

# Changelog

- new `ChooseProtocolSlideout` launches from the `RobotCard`'s overflow menu item, as well as the `RobotOverview`'s  main CTA
- `Slideout` should cover the entire app height, and it's `Overlay` should cover the entire app's width. 
-  clicking on the `Slideout`'s `Overlay` should always trigger the `onClose` callback
- `Slideout`'s children will scroll if overflowed, the footer and title will remain fixed.
 

# Review requests

- [ ] Launch a protocol from the choose protocol slideout on the device detail page
- [ ] Launch a protocol from the choose protocol slideout on the devices landing page

# Risk assessment
medium, this touches `Slideout`, `MenuItem`, and `StyledText` which all appear many times across the app.

### CODE REVIEW

File: app/src/assets/localization/en/devices_landing.json [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
   "modules": "Modules",
   "no_robots_found": "No robots found",
   "refresh": "Refresh",
-  "running": "running",
+  "restart_the_app": "Restart the app",
+  "restart_the_robot": "Restart the robot",
   "right_mount": "Right Mount",
+  "robot_settings": "robot settings",
+  "running": "running",
+  "run_protocol": "run protocol",
```
Comment (brenthagen) [inline app/src/assets/localization/en/devices_landing.json]: figma is "Run protocol"

File: app/src/organisms/Devices/RobotOverflowMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  const handleClickOutside: React.MouseEventHandler<HTMLDivElement> = e => {
+    e.preventDefault()
+    setShowOverflowMenu(false)
+  }
+  const handleClickConnectionTroubleshooting: React.MouseEventHandler<HTMLButtonElement> = e => {
+    e.preventDefault()
+    setShowConnectionTroubleshootingModal(true)
+  }
+  return (
+    <Flex
```
Comment (brenthagen) [inline app/src/organisms/Devices/RobotOverflowMenu.tsx:62]: clicking on the `ChooseProtocolSlideout` component or overlay triggers the wrapping `Link` for the `RobotCard` component and routes to the device detail page. I think an `onClick={e => e.preventDefault()}` on this outer `Flex` will resolve the issue. possibly sprinkling a few of these in the base overlay/slideout components would be preferable.

**Review by brenthagen — APPROVED** [review]
aside from the click event issue, works great!

File: app/src/organisms/Devices/RobotOverflowMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      <OverflowBtn alignSelf={ALIGN_FLEX_END} onClick={handleOverflowClick} />
+      {showOverflowMenu ? (
+        <Flex
+          width={SIZE_4}
+          zIndex={10}
+          borderRadius={'4px 4px 0px 0px'}
+          boxShadow={'0px 1px 3px rgba(0, 0, 0, 0.2)'}
+          position={POSITION_ABSOLUTE}
+          backgroundColor={COLORS.white}
+          top="3rem"
```
Comment (koji) [inline app/src/organisms/Devices/RobotOverflowMenu.tsx]: We can use `SPACING.spacing7`

File: app/src/organisms/Devices/ModuleCard/AboutModuleSlideout.tsx [hunk 4]
```diff
@@ -128,7 +127,6 @@ export const AboutModuleSlideout = (
         <Text
```
Comment (koji) [inline app/src/organisms/Devices/ModuleCard/AboutModuleSlideout.tsx]: Switching to `StyledText` would be good.

File: app/src/organisms/Devices/RobotOverview.tsx [hunk 5]
```diff
@@ -77,21 +83,28 @@ export function RobotOverview({
...
-          {/* this link will change once protocol selection designs are finalized and functionality built out */}
-          <Link
-            to={`/devices/${robot.name}/protocol-runs/run`}
-            id={`RobotOverview_runAProtocol`}
+          <PrimaryButton
+            textTransform={TEXT_TRANSFORM_NONE}
+            disabled={currentRunId != null || !isRobotViewable}
+            onClick={() => {
+              setShowChooseProtocolSlideout(true)
+              console.log('CLICKED\n\n\n')
```
Comment (koji) [inline app/src/organisms/Devices/RobotOverview.tsx]: Need to keep this `console.log`?

File: app/src/organisms/ChooseRobotSlideout/index.tsx [hunk 6]
```diff
@@ -44,6 +38,7 @@ import { AvailableRobotOption } from './AvailableRobotOption'
 
 import type { State, Dispatch } from '../../redux/types'
 import type { StyleProps } from '@opentrons/components'
```
Comment (koji) [inline app/src/organisms/ChooseRobotSlideout/index.tsx:40]: `@opentrons/components` is imported already, so probably moving this line into the above import statement would be good.

Comment (b-cooper) [reply app/src/organisms/ChooseRobotSlideout/index.tsx:40]: We try to separate out type imports from other named imports.  This sometimes means a file can have an `import` and a `import type` from the same location.

Comment (koji) [reply app/src/organisms/ChooseRobotSlideout/index.tsx:40]: oh okay, I missed `type`,

**Pull request merged**